recipes/libqasm/all/conanfile.py

```diff
@@ -0,0 +1,124 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMakeToolchain, CMakeDeps, CMake, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import copy, get, rm, rmdir
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.53.0"
+
+
+class LibqasmConan(ConanFile):
+    name = "libqasm"
+    license = "Apache-2.0"
+    homepage = "https://github.com/QuTech-Delft/libqasm"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Library to parse cQASM files"
+    topics = ("code generation", "parser", "compiler", "quantum compilation", "quantum simulation")
+    settings = "os", "compiler", "build_type", "arch"
+    package_type = "library"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_python": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_python": False
+    }
+
+    @property
+    def _should_build_test(self):
+        return not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+
+    @property
+    def _min_cppstd(self):
+        return "20"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "10",
+            "clang": "13",
+            "apple-clang": "14",
+            "Visual Studio": "16",
+            "msvc": "192"
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        # https://github.com/QuTech-Delft/libqasm/blob/0.6.3/src/CMakeLists.txt#L234
+        self.options["antlr4-cppruntime"].shared = self.options.shared
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def build_requirements(self):
+        self.tool_requires("tree-gen/1.0.7")
+        self.tool_requires("zulu-openjdk/11.0.19")
```

Comment:
```suggestion
        self.tool_requires("zulu-openjdk/21.0.1")
```
I assume the latest JDK version is not supported? It builds fine with it, though.

Reply:
Hm... not sure now why I'm sticking to 11.0.19. I'll give 21.0.1 a go in libqasm/0.6.4. Thanks!

Reply:
Done in libqasm/0.6.4!